<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="LegATo">
<title>LegATo Tutorial • LegATo</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="LegATo Tutorial">
<meta property="og:description" content="LegATo">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary" data-bs-theme="dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">LegATo</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../index.html">Home</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-tutorials">Tutorials</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-tutorials">
    <a class="dropdown-item" href="../articles/LegATo_vignette.html">Command Line Analysis</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Function Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Package News</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/aubreyodom/legato">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>LegATo Tutorial</h1>
                        <h4 data-toc-skip class="author">Aubrey R.
Odom</h4>
            <address class="author_afil">
      Program in Bioinformatics, Boston University, Boston,
MA<br><a class="author_email" href="mailto:#"></a><a href="mailto:aodom@bu.edu" class="email">aodom@bu.edu</a>
      </address>
                  
            <h4 data-toc-skip class="date">October 30, 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/wejlab/LegATo/blob/HEAD/vignettes/LegATo_vignette.Rmd" class="external-link"><code>vignettes/LegATo_vignette.Rmd</code></a></small>
      <div class="d-none name"><code>LegATo_vignette.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction-to-legato">Introduction to LegATo<a class="anchor" aria-label="anchor" href="#introduction-to-legato"></a>
</h2>
<div class="section level3">
<h3 id="streamlining-longitudinal-microbiome-profiling-in-bioconductor">Streamlining longitudinal microbiome profiling in Bioconductor<a class="anchor" aria-label="anchor" href="#streamlining-longitudinal-microbiome-profiling-in-bioconductor"></a>
</h3>
<p>Microbiome time-series data presents several distinct challenges,
including complex covariate dependencies and a variety of longitudinal
study designs. Furthermore, while several individual tools have been
created to aid in longitudinal microbiome data analysis, an
all-inclusive R-based toolkit has remained elusive.</p>
<p>To address these concerns, we have created LegATo (Longitudinal
mEtaGenomic Analysis TOolkit). LegATo is a suite of open-source software
tools for longitudinal microbiome analysis that is extendable to several
different study forms, all while promising optimal ease-of-use for
researchers. This toolkit will allow researchers to determine which
microbial taxa are affected over time by perturbations such as onset of
disease or lifestyle choices, and to predict the effects of these
perturbations over time, including changes in composition or stability
of commensal bacteria.</p>
<p>Currently, LegATo entertains a number of data cleaning, aggregation,
modeling and testing procedures. As we develop or learn of new methods,
we will add to the toolkit accordingly. We will soon add hierarchical
clustering tools and multivariate generalized estimating equations
(JGEEs) to adjust for the compositional nature of microbiome data.</p>
</div>
</div>
<div class="section level2">
<h2 id="getting-started-with-legato">Getting started with LegATo<a class="anchor" aria-label="anchor" href="#getting-started-with-legato"></a>
</h2>
<div class="section level3">
<h3 id="compatibility-with-multiassayexperiment-and-summarizedexperiment-objects">Compatibility with <code>MultiAssayExperiment</code> and
<code>SummarizedExperiment</code> objects<a class="anchor" aria-label="anchor" href="#compatibility-with-multiassayexperiment-and-summarizedexperiment-objects"></a>
</h3>
<p>The convenience of LegATo is in part credited to the integration of
<code>MultiAssayExperiment</code> and <code>SummarizedExperiment</code>
objects. These are reliable and clean data structures developed by the
as part of the <em><a href="https://bioconductor.org/packages/3.19/MultiAssayExperiment" class="external-link">MultiAssayExperiment</a></em>
and <em><a href="https://bioconductor.org/packages/3.19/SummarizedExperiment" class="external-link">SummarizedExperiment</a></em>
packages.</p>
<p>The <code>MultiAssayExperiment</code> container concisely stores a
<code>SummarizedExperiment</code> object, which aggregates data matrices
along with annotation information, metadata, and reduced dimensionality
data (PCA, t-SNE, etc.). To learn more about proper usage and context of
these objects, you may want to take a look at the <a href="https://bioconductor.org/packages/release/bioc/vignettes/MultiAssayExperiment/inst/doc/MultiAssayExperiment.html" class="external-link">MultiAssayExperiment
package vignette</a> and <a href="https://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html" class="external-link">SummarizedExperiment
package vignette</a>.</p>
<p>To install these two packages, run the following code:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"MultiAssayExperiment"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"SummarizedExperiment"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="installation-and-setup">Installation and setup<a class="anchor" aria-label="anchor" href="#installation-and-setup"></a>
</h3>
<p>To install the development version of LegATo from Github, run the
following code:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"wejlab/LegATo"</span><span class="op">)</span></span></code></pre></div>
<p>Now we’ll load the LegATo package into our library:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wejlab.github.io/LegATo-docs/" class="external-link">LegATo</a></span><span class="op">)</span></span></code></pre></div>
<p>For the purposes of this vignette, we’ll also load the
<code>ggeffects</code> and <code>emmeans</code> library:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://strengejacke.github.io/ggeffects/" class="external-link">ggeffects</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rvlenth.github.io/emmeans/" class="external-link">emmeans</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Welcome to emmeans.</span></span>
<span><span class="co">## Caution: You lose important information if you filter this package's results.</span></span>
<span><span class="co">## See '? untidy'</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="formatting-input-data-for-use-with-legato">Formatting input data for use with LegATo<a class="anchor" aria-label="anchor" href="#formatting-input-data-for-use-with-legato"></a>
</h2>
<p>LegATo functions expect data properly formatted in a
<code>MultiAssayExperiment</code> container, with naming conventions
that are compatible with <em><a href="https://bioconductor.org/packages/3.19/animalcules" class="external-link">animalcules</a></em>.
Obtaining properly formatted data can be achieved easily via the
<code><a href="../reference/create_formatted_MAE.html">create_formatted_MAE()</a></code> function from the LegATo package, as
detailed below.</p>
<div class="section level3">
<h3 id="a-note-on-compatibility-with-other-packages">A note on compatibility with other packages<a class="anchor" aria-label="anchor" href="#a-note-on-compatibility-with-other-packages"></a>
</h3>
<p>If you’ve analyzed your metagenomic or 16S data via the <a href="https://wejlab.github.io/metascope-docs/" class="external-link">MetaScope package</a>,
you can aggregate your sample outputs with user-input metadata using the
<code><a href="https://wejlab.github.io/metascope-docs/reference/convert_animalcules.html" class="external-link">MetaScope::convert_animalcules()</a></code> or
<code><a href="https://wejlab.github.io/metascope-docs/reference/convert_animalcules_silva.html" class="external-link">MetaScope::convert_animalcules_silva()</a></code> functions prior to
loading your data into LegATo. These functions create a taxonomy table
and collate all data into a LegATo- and animalcules-compatible
<code>MultiAssayExperiment</code> object.</p>
<p>You can also format your data analyzed elsewhere (e.g., PathoScope
2.0 or Kraken) with the <a href="https://wejlab.github.io/animalcules-docs/" class="external-link">animalcules
package</a> in the upload step with the R Shiny app, and then select
“Download Animalcules File” to obtain a <code>MAE</code> object that can
be used with LegATo.</p>
</div>
<div class="section level3">
<h3 id="requirements-for-using-create_formatted_mae">Requirements for using <code>create_formatted_MAE()</code><a class="anchor" aria-label="anchor" href="#requirements-for-using-create_formatted_mae"></a>
</h3>
<p>For the purposes of this example, we’ll look at some example inputs
unrelated to the main HIV-E infant data:</p>
<ul>
<li>The <code>counts</code> table is composed of raw microbial counts.
There is one row for each taxon, and one column for each sample.</li>
<li>The <code>tax</code> table is composed of taxonomy lineages. There
is one row for each taxon, and any number of columns each taxonomic
level (e.g., superkingdom, genus, species). The lowest level of taxonomy
serves as the rownames for the table, and is also stored in its own
column. Taxonomic levels increase in granularity from left to
right.</li>
<li>The <code>sample</code> table contains entries for each sample, with
unique sample IDs on the rows. The columns are metadata for the samples,
such as subjects or units on which repeated measures were taken,
groupings, pairings, and covariates.</li>
<li><code>rownames(counts) == rownames(tax)</code></li>
<li><code>colnames(counts) == colnames(sample)</code></li>
</ul>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"counts.csv"</span>, package <span class="op">=</span> <span class="st">"LegATo"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span>row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename_with</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html" class="external-link">str_replace</a></span><span class="op">(</span><span class="va">x</span>, <span class="st">"\\."</span>, <span class="st">"-"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tax</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"tax.csv"</span>, package <span class="op">=</span> <span class="st">"LegATo"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span>row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">sample</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"sample.csv"</span>, package <span class="op">=</span> <span class="st">"LegATo"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span>row.names <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p>Now we’ll look at the formatting of each:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ndim</span> <span class="op">&lt;-</span> <span class="fl">5</span></span>
<span><span class="va">counts</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="va">ndim</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="va">ndim</span><span class="op">)</span><span class="op">]</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>caption <span class="op">=</span> <span class="st">"Counts Table Preview"</span>,</span>
<span>               label <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Counts Table Preview</caption>
<thead><tr class="header">
<th align="left"></th>
<th align="right">X-1</th>
<th align="right">X-2</th>
<th align="right">X-3</th>
<th align="right">X-4</th>
<th align="right">X-5</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Acinetobacter_beijerinckii</td>
<td align="right">430</td>
<td align="right">18103</td>
<td align="right">810</td>
<td align="right">103</td>
<td align="right">74</td>
</tr>
<tr class="even">
<td align="left">Acinetobacter_bouvetii</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">Acinetobacter_guillouiae</td>
<td align="right">0</td>
<td align="right">5</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">Acinetobacter_gyllenbergii</td>
<td align="right">6</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">11</td>
</tr>
<tr class="odd">
<td align="left">Acinetobacter_indicus</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">9</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tax</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="va">ndim</span><span class="op">)</span>, <span class="op">]</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>caption <span class="op">=</span> <span class="st">"Taxonomy Table Preview"</span>,</span>
<span>               label <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Taxonomy Table Preview</caption>
<colgroup>
<col width="18%">
<col width="9%">
<col width="10%">
<col width="13%">
<col width="9%">
<col width="9%">
<col width="9%">
<col width="18%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="left">superkingdom</th>
<th align="left">phylum</th>
<th align="left">class</th>
<th align="left">order</th>
<th align="left">family</th>
<th align="left">genus</th>
<th align="left">species</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Acinetobacter_beijerinckii</td>
<td align="left">Bacteria</td>
<td align="left">Proteobacteria</td>
<td align="left">Gammaproteobacteria</td>
<td align="left">Moraxellales</td>
<td align="left">Moraxellaceae</td>
<td align="left">Acinetobacter</td>
<td align="left">Acinetobacter_beijerinckii</td>
</tr>
<tr class="even">
<td align="left">Acinetobacter_bouvetii</td>
<td align="left">Bacteria</td>
<td align="left">Proteobacteria</td>
<td align="left">Gammaproteobacteria</td>
<td align="left">Moraxellales</td>
<td align="left">Moraxellaceae</td>
<td align="left">Acinetobacter</td>
<td align="left">Acinetobacter_bouvetii</td>
</tr>
<tr class="odd">
<td align="left">Acinetobacter_guillouiae</td>
<td align="left">Bacteria</td>
<td align="left">Proteobacteria</td>
<td align="left">Gammaproteobacteria</td>
<td align="left">Moraxellales</td>
<td align="left">Moraxellaceae</td>
<td align="left">Acinetobacter</td>
<td align="left">Acinetobacter_guillouiae</td>
</tr>
<tr class="even">
<td align="left">Acinetobacter_gyllenbergii</td>
<td align="left">Bacteria</td>
<td align="left">Proteobacteria</td>
<td align="left">Gammaproteobacteria</td>
<td align="left">Moraxellales</td>
<td align="left">Moraxellaceae</td>
<td align="left">Acinetobacter</td>
<td align="left">Acinetobacter_gyllenbergii</td>
</tr>
<tr class="odd">
<td align="left">Acinetobacter_indicus</td>
<td align="left">Bacteria</td>
<td align="left">Proteobacteria</td>
<td align="left">Gammaproteobacteria</td>
<td align="left">Moraxellales</td>
<td align="left">Moraxellaceae</td>
<td align="left">Acinetobacter</td>
<td align="left">Acinetobacter_indicus</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sample</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="va">ndim</span><span class="op">)</span>, <span class="op">]</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>caption <span class="op">=</span> <span class="st">"Sample Table Preview"</span>,</span>
<span>               label <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Sample Table Preview</caption>
<thead><tr class="header">
<th align="left"></th>
<th align="left">Sample</th>
<th align="left">Subject</th>
<th align="left">Sex</th>
<th align="right">Month</th>
<th align="left">Group</th>
<th align="right">Pairing</th>
<th align="right">HairLength</th>
<th align="right">Age</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">X-1</td>
<td align="left">X-1</td>
<td align="left">S1</td>
<td align="left">Male</td>
<td align="right">1</td>
<td align="left">A</td>
<td align="right">1</td>
<td align="right">48.62753</td>
<td align="right">21.9</td>
</tr>
<tr class="even">
<td align="left">X-2</td>
<td align="left">X-2</td>
<td align="left">S2</td>
<td align="left">Male</td>
<td align="right">1</td>
<td align="left">B</td>
<td align="right">1</td>
<td align="right">53.53195</td>
<td align="right">67.7</td>
</tr>
<tr class="odd">
<td align="left">X-3</td>
<td align="left">X-3</td>
<td align="left">S3</td>
<td align="left">Female</td>
<td align="right">1</td>
<td align="left">A</td>
<td align="right">2</td>
<td align="right">47.61090</td>
<td align="right">78.9</td>
</tr>
<tr class="even">
<td align="left">X-4</td>
<td align="left">X-4</td>
<td align="left">S4</td>
<td align="left">Female</td>
<td align="right">1</td>
<td align="left">B</td>
<td align="right">2</td>
<td align="right">49.72870</td>
<td align="right">48.6</td>
</tr>
<tr class="odd">
<td align="left">X-5</td>
<td align="left">X-5</td>
<td align="left">S5</td>
<td align="left">Female</td>
<td align="right">1</td>
<td align="left">A</td>
<td align="right">3</td>
<td align="right">59.04509</td>
<td align="right">48.4</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="formatting-your-inputs-with-create_formatted_mae">Formatting your inputs with <code>create_formatted_MAE()</code><a class="anchor" aria-label="anchor" href="#formatting-your-inputs-with-create_formatted_mae"></a>
</h3>
<p>Once your data are formatted correctly, you can easily use
<code><a href="../reference/create_formatted_MAE.html">create_formatted_MAE()</a></code> like so:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">output</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_formatted_MAE.html">create_formatted_MAE</a></span><span class="op">(</span>counts_dat <span class="op">=</span> <span class="va">counts</span>,</span>
<span>                               tax_dat <span class="op">=</span> <span class="va">tax</span>,</span>
<span>                               metadata_dat <span class="op">=</span> <span class="va">sample</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/class.html" class="external-link">class</a></span><span class="op">(</span><span class="va">output</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "MultiAssayExperiment"</span></span>
<span><span class="co">## attr(,"package")</span></span>
<span><span class="co">## [1] "MultiAssayExperiment"</span></span></code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">MultiAssayExperiment</span><span class="fu">::</span><span class="fu">assays</span><span class="op">(</span><span class="va">output</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## List of length 1</span></span>
<span><span class="co">## names(1): MicrobeGenetics</span></span></code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a></span><span class="op">(</span><span class="va">output</span><span class="op">[[</span><span class="st">"MicrobeGenetics"</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## List of length 1</span></span>
<span><span class="co">## names(1): MGX</span></span></code></pre>
<p>If your data is in the format of a TreeSummarizedExperiment, you can
call <code>create_formatted_MAE(tree_SE)</code> to create a
<code>MAE</code> output that is compatible with the package.</p>
</div>
<div class="section level3">
<h3 id="adding-information-to-your-metadata-later-in-your-analysis">Adding information to your metadata later in your analysis<a class="anchor" aria-label="anchor" href="#adding-information-to-your-metadata-later-in-your-analysis"></a>
</h3>
<p>If information needs to be added to your data object at some later
point in the analysis, it is easiest to manipulate the raw data objects
(potentially via <code><a href="../reference/parse_MAE_SE.html">parse_MAE_SE()</a></code>) and then recreate the
<code>MAE</code> object with <code><a href="../reference/create_formatted_MAE.html">create_formatted_MAE()</a></code>.</p>
</div>
</div>
<div class="section level2">
<h2 id="example-data">Example data<a class="anchor" aria-label="anchor" href="#example-data"></a>
</h2>
<p>To illustrate the capabilities of LegATo, we will turn to a published
dataset from the following paper:</p>
<p>Odom AR, Gill CJ, Pieciak R et al. Characterization of longitudinal
nasopharyngeal microbiome patterns in maternally HIV-exposed Zambian
infants [version 2; peer review: 2 approved with reservations]. Gates
Open Res 2024, 6:143 (<a href="https://doi.org/10.12688/gatesopenres.14041.2" class="external-link uri">https://doi.org/10.12688/gatesopenres.14041.2</a>)</p>
<p>The raw dataset is archived in Zenodo:</p>
<p><em>Zenodo: Underlying data for ‘Characterization of longitudinal
nasopharyngeal microbiome patterns in maternally HIV-exposed Zambian
infants’. <a href="https://doi.org/10.5281/zenodo.725531324" class="external-link uri">https://doi.org/10.5281/zenodo.725531324</a></em></p>
<p>Further details on how the dataset was altered for inclusion in this
package are provided <a href="https://github.com/wejlab/LegATo/blob/main/inst/script/extdata_explanations.Rmd" class="external-link">here</a>.</p>
<div class="section level4">
<h4 id="example-data-context-and-structure">Example data context and structure<a class="anchor" aria-label="anchor" href="#example-data-context-and-structure"></a>
</h4>
<p>The example data consists of 167 NP swabs of healthy HIV-exposed,
uninfected (HEU; n=10) infants and their HIV(+) mothers and
HIV-unexposed, uninfected (HUU; n=10) infants and their HIV(-) mothers.
A total of 7 samples were collected per infant, with some missingness in
the data. These swabs were identified from a sample library collected in
Lusaka, Zambia between 2015 and 2016.</p>
<p>The analysis objective is to parse the association between the NP
resident bacteria and infant HIV exposure during the first 3.5 months
(14 weeks) of life, a critical time in microbiome maturation.</p>
<p>All data was processed with <code>PathoScope 2.0</code>, and the
sample outputs were aggregated with the <code>animalcules</code> R
package.</p>
<p>For this analysis, we will work with the infant data only and subset
the data accordingly.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"MAE.RDS"</span>, package <span class="op">=</span> <span class="st">"LegATo"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">dat_subsetted</span> <span class="op">&lt;-</span> <span class="fu">MultiAssayExperiment</span><span class="fu">::</span><span class="fu"><a href="https://github.com/waldronlab/MultiAssayExperiment/reference/subsetBy.html" class="external-link">subsetByColData</a></span><span class="op">(</span><span class="va">dat</span>,</span>
<span>                                                       <span class="va">dat</span><span class="op">$</span><span class="va">MothChild</span> <span class="op">==</span> <span class="st">"Infant"</span><span class="op">)</span></span></code></pre></div>
<p>This leaves us with 129 samples in our analysis.</p>
</div>
</div>
<div class="section level2">
<h2 id="initial-data-exploration">Initial data exploration<a class="anchor" aria-label="anchor" href="#initial-data-exploration"></a>
</h2>
<div class="section level3">
<h3 id="multiassayexperiment-mae-manipulation">
<code>MultiAssayExperiment (MAE)</code> manipulation<a class="anchor" aria-label="anchor" href="#multiassayexperiment-mae-manipulation"></a>
</h3>
<p><code>LegATo</code> has several functions for manipulating
<code>MAE</code> objects, delineated below:</p>
<div class="section level4">
<h4 id="clean-up-animalcules-formatted-maes">Clean up animalcules-formatted <code>MAE</code>s<a class="anchor" aria-label="anchor" href="#clean-up-animalcules-formatted-maes"></a>
</h4>
<p>If your data were aggregated via the <code>animalcules</code>
package, as was done for the example data, the samples were aggregated
at the strain taxonomic level and have residual taxonomy IDs present.
The <code><a href="../reference/clean_MAE.html">clean_MAE()</a></code> function cleans up data suffering from
these issues.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dat_cleaned</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/clean_MAE.html">clean_MAE</a></span><span class="op">(</span><span class="va">dat_subsetted</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Registered S3 method overwritten by 'rmutil':</span></span>
<span><span class="co">##   method         from</span></span>
<span><span class="co">##   print.response httr</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="filter-mae">Filter MAE<a class="anchor" aria-label="anchor" href="#filter-mae"></a>
</h4>
<p>Many metagenomic pipelines identify taxon abundances at extremely
small levels, which can be noisy to deal with in an analysis. The
<code>filter_MAE</code> function smoothly transforms reads belonging to
taxa with an overall genera threshold under the <code>filter_prop</code>
(filter proportion) argument, which we will set as 0.005.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dat_filt_1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_MAE.html">filter_MAE</a></span><span class="op">(</span><span class="va">dat_cleaned</span>, relabu_threshold <span class="op">=</span> <span class="fl">0.05</span>, occur_pct_cutoff <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## The overall range of relative abundance counts between samples is (13218, 3016276)</span></span></code></pre>
<pre><code><span><span class="co">## Number of OTUs that exhibit a relative abundance &gt;0.05% in at least 10% of the total samples: 68/1136</span></span></code></pre>
<p>The <code>filter_animalcules_MAE</code> function utilizes the same
filtering mechanism as in the animalcules package:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dat_filt</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_animalcules_MAE.html">filter_animalcules_MAE</a></span><span class="op">(</span><span class="va">dat_cleaned</span>, <span class="fl">0.05</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="parse-mae-to-extract-data">Parse MAE to extract data<a class="anchor" aria-label="anchor" href="#parse-mae-to-extract-data"></a>
</h4>
<p>If you want to take a closer look at your data, you can easily
extract it into the counts, taxonomy, and sample metadata tables using
the <code><a href="../reference/parse_MAE_SE.html">parse_MAE_SE()</a></code> function.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">parsed</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/parse_MAE_SE.html">parse_MAE_SE</a></span><span class="op">(</span><span class="va">dat_filt</span>, which_assay <span class="op">=</span> <span class="st">"MicrobeGenetics"</span>, type <span class="op">=</span> <span class="st">"MAE"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">parsed</span><span class="op">$</span><span class="va">counts</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">5</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">5</span><span class="op">)</span><span class="op">]</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>caption <span class="op">=</span> <span class="st">"Counts Table"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Counts Table</caption>
<thead><tr class="header">
<th align="left"></th>
<th align="right">X1755</th>
<th align="right">X2216</th>
<th align="right">X2431</th>
<th align="right">X2561</th>
<th align="right">X2699</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Corynebacterium accolens</td>
<td align="right">0</td>
<td align="right">9643</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">43435</td>
</tr>
<tr class="even">
<td align="left">Corynebacterium ammoniagenes</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">Corynebacterium appendicis</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">Corynebacterium argentoratense</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">4</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">Corynebacterium aurimucosum</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">58</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">parsed</span><span class="op">$</span><span class="va">sam</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">5</span><span class="op">)</span>, <span class="op">]</span>  <span class="op">|&gt;</span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>caption <span class="op">=</span> <span class="st">"Sample Metadata"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Sample Metadata</caption>
<thead><tr class="header">
<th align="left"></th>
<th align="left">Sample</th>
<th align="left">Subject</th>
<th align="left">HIVStatus</th>
<th align="left">MothChild</th>
<th align="right">timepoint</th>
<th align="right">Age</th>
<th align="right">pairing</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">X1755</td>
<td align="left">X1755</td>
<td align="left">0469-1</td>
<td align="left">Control</td>
<td align="left">Infant</td>
<td align="right">0</td>
<td align="right">7</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">X2216</td>
<td align="left">X2216</td>
<td align="left">0507-1</td>
<td align="left">Control</td>
<td align="left">Infant</td>
<td align="right">0</td>
<td align="right">6</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">X2431</td>
<td align="left">X2431</td>
<td align="left">0539-1</td>
<td align="left">Control</td>
<td align="left">Infant</td>
<td align="right">0</td>
<td align="right">9</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="left">X2561</td>
<td align="left">X2561</td>
<td align="left">0554-1</td>
<td align="left">HIV</td>
<td align="left">Infant</td>
<td align="right">0</td>
<td align="right">6</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">X2699</td>
<td align="left">X2699</td>
<td align="left">0620-1</td>
<td align="left">HIV</td>
<td align="left">Infant</td>
<td align="right">0</td>
<td align="right">4</td>
<td align="right">2</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">parsed</span><span class="op">$</span><span class="va">tax</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fl">5</span><span class="op">)</span>, <span class="op">]</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>caption <span class="op">=</span> <span class="st">"Taxonomy Table"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Taxonomy Table</caption>
<colgroup>
<col width="19%">
<col width="8%">
<col width="9%">
<col width="8%">
<col width="11%">
<col width="12%">
<col width="10%">
<col width="19%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="left">superkingdom</th>
<th align="left">phylum</th>
<th align="left">class</th>
<th align="left">order</th>
<th align="left">family</th>
<th align="left">genus</th>
<th align="left">species</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Corynebacterium accolens</td>
<td align="left">Bacteria</td>
<td align="left">Actinobacteria</td>
<td align="left">Actinomycetia</td>
<td align="left">Corynebacteriales</td>
<td align="left">Corynebacteriaceae</td>
<td align="left">Corynebacterium</td>
<td align="left">Corynebacterium accolens</td>
</tr>
<tr class="even">
<td align="left">Corynebacterium ammoniagenes</td>
<td align="left">Bacteria</td>
<td align="left">Actinobacteria</td>
<td align="left">Actinomycetia</td>
<td align="left">Corynebacteriales</td>
<td align="left">Corynebacteriaceae</td>
<td align="left">Corynebacterium</td>
<td align="left">Corynebacterium ammoniagenes</td>
</tr>
<tr class="odd">
<td align="left">Corynebacterium appendicis</td>
<td align="left">Bacteria</td>
<td align="left">Actinobacteria</td>
<td align="left">Actinomycetia</td>
<td align="left">Corynebacteriales</td>
<td align="left">Corynebacteriaceae</td>
<td align="left">Corynebacterium</td>
<td align="left">Corynebacterium appendicis</td>
</tr>
<tr class="even">
<td align="left">Corynebacterium argentoratense</td>
<td align="left">Bacteria</td>
<td align="left">Actinobacteria</td>
<td align="left">Actinomycetia</td>
<td align="left">Corynebacteriales</td>
<td align="left">Corynebacteriaceae</td>
<td align="left">Corynebacterium</td>
<td align="left">Corynebacterium argentoratense</td>
</tr>
<tr class="odd">
<td align="left">Corynebacterium aurimucosum</td>
<td align="left">Bacteria</td>
<td align="left">Actinobacteria</td>
<td align="left">Actinomycetia</td>
<td align="left">Corynebacteriales</td>
<td align="left">Corynebacteriaceae</td>
<td align="left">Corynebacterium</td>
<td align="left">Corynebacterium aurimucosum</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="summarizing-microbial-counts-data">Summarizing microbial counts data<a class="anchor" aria-label="anchor" href="#summarizing-microbial-counts-data"></a>
</h4>
<p>It is fairly convenient to summarize the average number of reads for
a report with <code><a href="../reference/get_summary_table.html">get_summary_table()</a></code>. The table groups by
user-provided discrete covariates.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">group_vars</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HIVStatus"</span>, <span class="st">"MothChild"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/get_summary_table.html">get_summary_table</a></span><span class="op">(</span><span class="va">dat_filt</span>, <span class="va">group_vars</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>caption <span class="op">=</span> <span class="st">"Summary Table"</span>, label <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Summary Table</caption>
<colgroup>
<col width="14%">
<col width="14%">
<col width="15%">
<col width="14%">
<col width="14%">
<col width="14%">
<col width="14%">
</colgroup>
<thead><tr class="header">
<th align="left">HIVStatus</th>
<th align="left">MothChild</th>
<th align="right">mean_reads</th>
<th align="right">sd_reads</th>
<th align="right">min_reads</th>
<th align="right">max_reads</th>
<th align="right">num_total</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Control</td>
<td align="left">Infant</td>
<td align="right">156209.2</td>
<td align="right">408000.52</td>
<td align="right">24176</td>
<td align="right">3016276</td>
<td align="right">68</td>
</tr>
<tr class="even">
<td align="left">HIV</td>
<td align="left">Infant</td>
<td align="right">90008.2</td>
<td align="right">51395.08</td>
<td align="right">13218</td>
<td align="right">202079</td>
<td align="right">61</td>
</tr>
</tbody>
</table>
<p>The <code>get_top_taxa</code> function outputs a
<code>data.frame</code> that lists taxa in order of relative
abundance.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">best_genus</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_top_taxa.html">get_top_taxa</a></span><span class="op">(</span><span class="va">dat_filt</span>, <span class="st">"genus"</span><span class="op">)</span></span>
<span><span class="va">best_genus</span> <span class="op">|&gt;</span> <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>caption <span class="op">=</span> <span class="st">"Table of genera, ranked by abundance"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>Table of genera, ranked by abundance</caption>
<thead><tr class="header">
<th align="left">taxon</th>
<th align="right">allmeans</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Dolosigranulum</td>
<td align="right">0.1892754</td>
</tr>
<tr class="even">
<td align="left">Streptococcus</td>
<td align="right">0.1730879</td>
</tr>
<tr class="odd">
<td align="left">Moraxella</td>
<td align="right">0.1713469</td>
</tr>
<tr class="even">
<td align="left">Staphylococcus</td>
<td align="right">0.1554891</td>
</tr>
<tr class="odd">
<td align="left">Corynebacterium</td>
<td align="right">0.1290279</td>
</tr>
<tr class="even">
<td align="left">Other</td>
<td align="right">0.0979779</td>
</tr>
<tr class="odd">
<td align="left">Haemophilus</td>
<td align="right">0.0837949</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="other-data-manipulation-functions">Other data manipulation functions<a class="anchor" aria-label="anchor" href="#other-data-manipulation-functions"></a>
</h4>
<p>If you want to conduct your own analyses, the
<code><a href="../reference/get_long_data.html">get_long_data()</a></code> function will prove convenient. The
<code><a href="../reference/get_stacked_data.html">get_stacked_data()</a></code> function can be used for certain
visualizations that utilize a relative abundance aggregation
approach.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">longdat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_long_data.html">get_long_data</a></span><span class="op">(</span><span class="va">dat_filt</span>, <span class="st">"genus"</span>, log <span class="op">=</span> <span class="cn">TRUE</span>, counts_to_CPM <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">stackeddat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_stacked_data.html">get_stacked_data</a></span><span class="op">(</span><span class="va">dat_filt</span>, <span class="st">"genus"</span>, covariate_1 <span class="op">=</span> <span class="st">"HIVStatus"</span>,</span>
<span>                               covariate_time <span class="op">=</span> <span class="st">"timepoint"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="visualizing-data">Visualizing Data<a class="anchor" aria-label="anchor" href="#visualizing-data"></a>
</h3>
<p>There are several plots by which we can visualize changes in relative
abundance over time, accounting for a given covariate. In this case, we
are interested in HIV exposure.</p>
<p>We’ll select a palette using <code>paletteer</code>.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">this_palette</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#709AE1"</span>, <span class="st">"#8A9197"</span>, <span class="st">"#D2AF81"</span>, <span class="st">"#FD7446"</span>, <span class="st">"#D5E4A2"</span>, <span class="st">"#197EC0"</span>, <span class="st">"#F05C3B"</span>, <span class="st">"#46732E"</span>,</span>
<span>                  <span class="st">"#71D0F5"</span>, <span class="st">"#370335"</span>, <span class="st">"#075149"</span>, <span class="st">"#C80813"</span>, <span class="st">"#91331F"</span>, <span class="st">"#1A9993"</span>, <span class="st">"#FD8CC1"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">magrittr</span><span class="fu">::</span><span class="fu"><a href="https://magrittr.tidyverse.org/reference/aliases.html" class="external-link">extract</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_len</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">best_genus</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="alluvial-plots">Alluvial plots<a class="anchor" aria-label="anchor" href="#alluvial-plots"></a>
</h4>
<p>Alluvial diagrams illustrate individual taxa as stream fields that
change position at different time points. The height of a stream field
represents the relative abundance of that taxon. At a given time point,
stream fields are ranked from the highest to lowest abundance (top to
bottom). These can be constructed with <code><a href="../reference/plot_alluvial.html">plot_alluvial()</a></code>.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_alluvial.html">plot_alluvial</a></span><span class="op">(</span>dat <span class="op">=</span> <span class="va">dat_filt</span>, </span>
<span>              taxon_level <span class="op">=</span> <span class="st">"genus"</span>, </span>
<span>              covariate_1 <span class="op">=</span> <span class="st">"HIVStatus"</span>,</span>
<span>              covariate_time <span class="op">=</span> <span class="st">"timepoint"</span>,</span>
<span>              palette_input <span class="op">=</span> <span class="va">this_palette</span>,</span>
<span>              subtitle <span class="op">=</span> <span class="st">"Alluvial plot"</span><span class="op">)</span></span></code></pre></div>
<p><img src="LegATo_vignette_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
<p>In context, this plot indicates a high abundance of Staphylococcus in
both the control and HIV-E infants in the first two weeks of life, but
other genera quickly overtake <strong>Staphylococcus</strong> in
dominance as shown by the changing rank of the streams for each time
point. Overall, <strong>Streptococcus</strong> seems to have a far
greater presence in the HIV-E infants’ nasopharyngeal microbiota</p>
</div>
<div class="section level4">
<h4 id="spaghetti-plots">Spaghetti plots<a class="anchor" aria-label="anchor" href="#spaghetti-plots"></a>
</h4>
<p>We can create spaghetti or volatility plots to elucidate changes over
time on a sample level for a given taxon. This is advantageous as other
visualization methods are often aggregates of multiple samples and lack
granularity. These plots can be created with
<code><a href="../reference/plot_spaghetti.html">plot_spaghetti()</a></code>.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span>  <span class="fu"><a href="../reference/plot_spaghetti.html">plot_spaghetti</a></span><span class="op">(</span>dat <span class="op">=</span> <span class="va">dat_filt</span>,</span>
<span>               covariate_time <span class="op">=</span> <span class="st">"timepoint"</span>,</span>
<span>               covariate_1 <span class="op">=</span> <span class="st">"HIVStatus"</span>,</span>
<span>               unit_var <span class="op">=</span> <span class="st">"Subject"</span>,</span>
<span>               taxon_level <span class="op">=</span> <span class="st">"genus"</span>,</span>
<span>               which_taxon <span class="op">=</span> <span class="st">"Staphylococcus"</span>,</span>
<span>               palette_input <span class="op">=</span> <span class="va">this_palette</span>,</span>
<span>               title <span class="op">=</span> <span class="st">"Spaghetti Plot"</span>,</span>
<span>               subtitle <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"Infant Age (timepoint)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Relative Abundance (log CPM)"</span><span class="op">)</span></span></code></pre></div>
<p><img src="LegATo_vignette_files/figure-html/unnamed-chunk-16-1.png" width="700"></p>
<p>Here we plot changes in the <strong>Staphylococcus</strong> log
counts per million (CPM) across time points. The spaghetti plot format
clues us into the few outlier infants in the control group that seem to
have extremely high counts in the first few weeks of life.</p>
</div>
<div class="section level4">
<h4 id="stacked-bar-plots">Stacked bar plots<a class="anchor" aria-label="anchor" href="#stacked-bar-plots"></a>
</h4>
<p>Stacked bar plots are used here to visualize the relative abundance
of microbes at a given taxonomic level in each sample, represented as a
single bar, labeled by time point, and plotted within each HIV exposure
status group for separate mothers and infant comparisons. Use
<code>plot_stacked_bar</code>.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_stacked_bar.html">plot_stacked_bar</a></span><span class="op">(</span><span class="va">dat_filt</span>, <span class="st">"genus"</span>, </span>
<span>             <span class="st">"HIVStatus"</span>,</span>
<span>             <span class="st">"timepoint"</span>,</span>
<span>             palette_input <span class="op">=</span> <span class="va">this_palette</span><span class="op">)</span></span></code></pre></div>
<p><img src="LegATo_vignette_files/figure-html/unnamed-chunk-17-1.png" width="700"></p>
<p>We see from the stacked bar plots that many of the microbes seem to
maintain similar trends in both control and HIV-E infants over time,
with <strong>Streptococcus</strong> producing slightly higher abundances
in the HIV-E group.</p>
</div>
<div class="section level4">
<h4 id="stacked-area-charts">Stacked area charts<a class="anchor" aria-label="anchor" href="#stacked-area-charts"></a>
</h4>
<p>Stacked area charts are similar to stacked bar plots, but provide for
continuity between time points. These are created with
<code><a href="../reference/plot_stacked_area.html">plot_stacked_area()</a></code>.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_stacked_area.html">plot_stacked_area</a></span><span class="op">(</span><span class="va">dat_filt</span>, <span class="st">"genus"</span>, </span>
<span>                  <span class="st">"HIVStatus"</span>,</span>
<span>                  <span class="st">"timepoint"</span>,</span>
<span>                  palette_input <span class="op">=</span> <span class="va">this_palette</span><span class="op">)</span></span></code></pre></div>
<p><img src="LegATo_vignette_files/figure-html/unnamed-chunk-18-1.png" width="700"></p>
<p>The stacked area chart is offers a similar method of visualizing the
data.</p>
</div>
<div class="section level4">
<h4 id="heatmaps">Heatmaps<a class="anchor" aria-label="anchor" href="#heatmaps"></a>
</h4>
<p>With <code>plot_heatmap</code>, you can plot a heatmap of a specific
microbe to determine changes along one or more covariates.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">this_taxon</span> <span class="op">&lt;-</span> <span class="va">parsed</span><span class="op">$</span><span class="va">counts</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">animalcules</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/animalcules/man/upsample_counts.html" class="external-link">upsample_counts</a></span><span class="op">(</span><span class="va">parsed</span><span class="op">$</span><span class="va">tax</span>, <span class="st">"genus"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">animalcules</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/animalcules/man/counts_to_logcpm.html" class="external-link">counts_to_logcpm</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_heatmap.html">plot_heatmap</a></span><span class="op">(</span>inputData <span class="op">=</span> <span class="va">this_taxon</span>,</span>
<span>                  annotationData <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">parsed</span><span class="op">$</span><span class="va">sam</span>, <span class="st">"timepoint"</span>, <span class="st">"HIVStatus"</span>, <span class="st">"pairing"</span><span class="op">)</span>,</span>
<span>                  name <span class="op">=</span> <span class="st">"Data"</span>,</span>
<span>                  plot_title <span class="op">=</span> <span class="st">"Example"</span>,</span>
<span>                  plottingColNames <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                  annotationColNames <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                  colList <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>                  scale <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                  showColumnNames <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                  showRowNames <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                  colorSets <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Set1"</span>, <span class="st">"Set2"</span>, <span class="st">"Set3"</span>, <span class="st">"Pastel1"</span>, <span class="st">"Pastel2"</span>, <span class="st">"Accent"</span>, <span class="st">"Dark2"</span>,</span>
<span>                                <span class="st">"Paired"</span><span class="op">)</span>,</span>
<span>                  choose_color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="st">"gray95"</span>, <span class="st">"red"</span><span class="op">)</span>,</span>
<span>                  split_heatmap <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>                  column_order <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: The input is a data frame-like object, convert it to a matrix.</span></span></code></pre>
<p><img src="LegATo_vignette_files/figure-html/unnamed-chunk-19-1.png" width="700"></p>
<p>From the heatmap, we find that clustering does not seem to find a
clear relationship between genera (on the rows) and samples (on the
columns). Three annotations were plotted on the top bar: time point, HIV
status, and pairing.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="longitudinal-data-analysis">Longitudinal data analysis<a class="anchor" aria-label="anchor" href="#longitudinal-data-analysis"></a>
</h2>
<p>For a concrete analysis of longitudinal microbiome data,
<code>LegATo</code> provides NMIT, both a paired and unpaired
multivariate Hotelling’s T-squared test, generalized estimating
equations (GEEs) and linear mixed effects models (LMEMs/LMMs).</p>
<p>A brief overview of each method is provided below.</p>
<div class="section level3">
<h3 id="nonparametric-microbial-interdependence-test-nmit">Nonparametric microbial interdependence test (NMIT)<a class="anchor" aria-label="anchor" href="#nonparametric-microbial-interdependence-test-nmit"></a>
</h3>
<p>NMIT is a multivariate distance-based test intended to evaluate the
association between key phenotypic variables and microbial
interdependence. The test determines longitudinal sample similarity as a
function of temporal microbial composition.</p>
<p>The authors thank Yilong Zhang for providing his code for adaptation
into LegATo.</p>
<p>Citations:</p>
<p>Yilong Zhang, Sung Won Han, Laura M Cox, and Huilin Li. A
multivariate distance-based analytic framework for microbial
interdependence association test in longitudinal study. Genetic
epidemiology, 41(8):769–778, 2017. <a href="doi:10.1002/gepi.22065" class="uri">doi:10.1002/gepi.22065</a>.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dat_1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_MAE.html">filter_MAE</a></span><span class="op">(</span><span class="va">dat_cleaned</span>, <span class="fl">0.05</span>, <span class="fl">10</span>, <span class="st">"species"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## The overall range of relative abundance counts between samples is (13218, 3016276)</span></span></code></pre>
<pre><code><span><span class="co">## Number of OTUs that exhibit a relative abundance &gt;0.05% in at least 10% of the total samples: 46/1136</span></span></code></pre>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/NMIT.html">NMIT</a></span><span class="op">(</span><span class="va">dat_1</span>, unit_var <span class="op">=</span> <span class="st">"Subject"</span>, fixed_cov <span class="op">=</span> <span class="st">"HIVStatus"</span>,</span>
<span>     covariate_time <span class="op">=</span> <span class="st">"timepoint"</span>,</span>
<span>     method <span class="op">=</span> <span class="st">"kendall"</span>, dist_type <span class="op">=</span> <span class="st">"F"</span>,</span>
<span>     heatmap <span class="op">=</span> <span class="cn">TRUE</span>, classify <span class="op">=</span> <span class="cn">FALSE</span>, fill_na <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in stats::cor(x = df, y, method = method): the standard deviation is</span></span>
<span><span class="co">## zero</span></span></code></pre>
<p><img src="LegATo_vignette_files/figure-html/unnamed-chunk-20-1.png" width="700"></p>
<pre><code><span><span class="co">## NULL</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="hotellings-t2-test">Hotelling’s T^2 Test<a class="anchor" aria-label="anchor" href="#hotellings-t2-test"></a>
</h3>
<p>Hotelling’s T2 tests can be used to determine whether the microbiome
profiles exhibit notable differences or trends across time and groups.
We may then use t-tests to identify which genera contributed most to
these differences.</p>
<p>For this example, HEU and HUU infants are designated as the two
sampling units on which the relative abundances of the p most abundant
genera will be measured.</p>
<p>For paired tests, we chose p = 6 variables to ensure that n &lt; p so
that singularity could be avoided and T2 could be properly computed,
where n is the number of measurements in a sampling unit. Normality is
met by using microbe abundances in log CPM units, which is calculated
within the function.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/test_hotelling_t2.html">test_hotelling_t2</a></span><span class="op">(</span>dat <span class="op">=</span> <span class="va">dat_1</span>,</span>
<span>                  test_index <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">dat_filt</span><span class="op">$</span><span class="va">MothChild</span> <span class="op">==</span> <span class="st">"Infant"</span> <span class="op">&amp;</span></span>
<span>                                       <span class="va">dat_filt</span><span class="op">$</span><span class="va">timepoint</span> <span class="op">==</span> <span class="fl">6</span><span class="op">)</span>,</span>
<span>                  taxon_level <span class="op">=</span> <span class="st">"genus"</span>,</span>
<span>                  num_taxa <span class="op">=</span> <span class="fl">6</span>,</span>
<span>                  paired <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                  grouping_var <span class="op">=</span> <span class="st">"HIVStatus"</span>,</span>
<span>                  pairing_var <span class="op">=</span> <span class="st">"pairing"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $df1</span></span>
<span><span class="co">## [1] 6</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $df2</span></span>
<span><span class="co">## [1] 2</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $crit_F</span></span>
<span><span class="co">## [1] 19.32953</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $F_stat</span></span>
<span><span class="co">## [1] 2.665069</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $pvalue</span></span>
<span><span class="co">## [1] 0.2978084</span></span></code></pre>
<p>Group comparisons can also be tested on unpaired data:</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/test_hotelling_t2.html">test_hotelling_t2</a></span><span class="op">(</span>dat <span class="op">=</span> <span class="va">dat_1</span>,</span>
<span>                  test_index <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">dat_filt</span><span class="op">$</span><span class="va">timepoint</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span>,</span>
<span>                  taxon_level <span class="op">=</span> <span class="st">"genus"</span>,</span>
<span>                  num_taxa <span class="op">=</span> <span class="fl">6</span>,</span>
<span>                  grouping_var <span class="op">=</span> <span class="st">"HIVStatus"</span>,</span>
<span>                  unit_var <span class="op">=</span> <span class="st">"Subject"</span>,</span>
<span>                  paired <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $df1</span></span>
<span><span class="co">## [1] 6</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $df2</span></span>
<span><span class="co">## [1] 13</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $crit_F</span></span>
<span><span class="co">## [1] 2.915269</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $F_stat</span></span>
<span><span class="co">## [1] 0.7611613</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $pvalue</span></span>
<span><span class="co">## [1] 0.6127997</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="modeling">Modeling<a class="anchor" aria-label="anchor" href="#modeling"></a>
</h3>
<div class="section level4">
<h4 id="generalized-estimating-equations-gees">Generalized Estimating Equations (GEEs)<a class="anchor" aria-label="anchor" href="#generalized-estimating-equations-gees"></a>
</h4>
<p>Generalized estimating equations (GEEs) as described in Liang and
Zeger (1986) and extended by Agresti (2002) have been widely used for
modeling longitudinal data, and more recently for longitudinal
microbiome data.</p>
<p>For each genus present in the microbial aggregate of samples, we
model normalized log CPM relative taxon counts, estimating the effects
of time point and HIV exposure status and their interaction, while
accounting for the underlying structure of clusters formed by individual
subjects.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">output</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/run_gee_model.html">run_gee_model</a></span><span class="op">(</span><span class="va">dat_1</span>, unit_var <span class="op">=</span> <span class="st">"Subject"</span>,</span>
<span>                        fixed_cov <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HIVStatus"</span>, <span class="st">"timepoint"</span><span class="op">)</span>,</span>
<span>                        corstr <span class="op">=</span> <span class="st">"ar1"</span>,</span>
<span>                        plot_out <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                        plotsave_loc <span class="op">=</span> <span class="st">"."</span>,</span>
<span>                        plot_terms <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">output</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>caption <span class="op">=</span> <span class="st">"GEE Outputs"</span><span class="op">)</span></span></code></pre></div>
<table style="width:100%;" class="table">
<caption>GEE Outputs</caption>
<colgroup>
<col width="9%">
<col width="16%">
<col width="11%">
<col width="7%">
<col width="11%">
<col width="10%">
<col width="10%">
<col width="12%">
<col width="9%">
</colgroup>
<thead><tr class="header">
<th align="left">Coefficient</th>
<th align="right">Coefficient Estimate</th>
<th align="right">Standard Error</th>
<th align="right">Statistic</th>
<th align="right">Unadj p-value</th>
<th align="right">Lower 95% CI</th>
<th align="right">Upper 95% CI</th>
<th align="left">Taxon</th>
<th align="right">Adj p-value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">(Intercept)</td>
<td align="right">54987.56051</td>
<td align="right">45931.419</td>
<td align="right">1.4332079</td>
<td align="right">0.2312416</td>
<td align="right">-35036.366</td>
<td align="right">145011.487</td>
<td align="left">Dolosigranulum</td>
<td align="right">0.2654996</td>
</tr>
<tr class="even">
<td align="left">(Intercept)</td>
<td align="right">4566.07283</td>
<td align="right">3252.186</td>
<td align="right">1.9712188</td>
<td align="right">0.1603186</td>
<td align="right">-1808.095</td>
<td align="right">10940.241</td>
<td align="left">Streptococcus</td>
<td align="right">0.2471246</td>
</tr>
<tr class="odd">
<td align="left">(Intercept)</td>
<td align="right">10783.40922</td>
<td align="right">4927.862</td>
<td align="right">4.7884507</td>
<td align="right">0.0286512</td>
<td align="right">1124.976</td>
<td align="right">20441.842</td>
<td align="left">Moraxella</td>
<td align="right">0.1637734</td>
</tr>
<tr class="even">
<td align="left">(Intercept)</td>
<td align="right">59258.78618</td>
<td align="right">26143.122</td>
<td align="right">5.1379585</td>
<td align="right">0.0234082</td>
<td align="right">8019.208</td>
<td align="right">110498.365</td>
<td align="left">Staphylococcus</td>
<td align="right">0.1637734</td>
</tr>
<tr class="odd">
<td align="left">(Intercept)</td>
<td align="right">46202.05734</td>
<td align="right">20777.660</td>
<td align="right">4.9445795</td>
<td align="right">0.0261726</td>
<td align="right">5478.592</td>
<td align="right">86925.522</td>
<td align="left">Corynebacterium</td>
<td align="right">0.1637734</td>
</tr>
<tr class="even">
<td align="left">(Intercept)</td>
<td align="right">54.48054</td>
<td align="right">1827.981</td>
<td align="right">0.0008883</td>
<td align="right">0.9762236</td>
<td align="right">-3528.297</td>
<td align="right">3637.258</td>
<td align="left">Haemophilus</td>
<td align="right">0.9762236</td>
</tr>
</tbody>
</table>
<p>You can also create plots of the covariates, which will be saved to a
folder specified by the user:</p>
<pre><code><span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span>
<span><span class="co">## Warning: Some of the focal terms are of type `character`. This may lead to</span></span>
<span><span class="co">##   unexpected results. It is recommended to convert these variables to</span></span>
<span><span class="co">##   factors before fitting the model.</span></span>
<span><span class="co">##   The following variables in are of type character: `HIVStatus`</span></span></code></pre>
<pre><code><span><span class="co">## [1] "Abiotrophia_HIVStatus.png"   "Abiotrophia_timepoint.png"  </span></span>
<span><span class="co">## [3] "Acinetobacter_HIVStatus.png" "Acinetobacter_timepoint.png"</span></span>
<span><span class="co">## [5] "Aeromonas_HIVStatus.png"     "Aeromonas_timepoint.png"</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="linear-mixed-models">Linear Mixed Models<a class="anchor" aria-label="anchor" href="#linear-mixed-models"></a>
</h4>
<p>Similarly, you can also run linear mixed-effects models:</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># If there are issues with matrix being positive definite</span></span>
<span><span class="co"># Revisit filtering parameters in filter_MAE</span></span>
<span><span class="va">output</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/run_lmm_model.html">run_lmm_model</a></span><span class="op">(</span><span class="va">dat_1</span>, unit_var <span class="op">=</span> <span class="st">"Subject"</span>,</span>
<span>                        taxon_level <span class="op">=</span> <span class="st">"genus"</span>,</span>
<span>                        fixed_cov <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"timepoint"</span>, <span class="st">"HIVStatus"</span><span class="op">)</span>,</span>
<span>                        plot_out <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                        plotsave_loc <span class="op">=</span> <span class="st">"."</span>,</span>
<span>                        plot_terms <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## boundary (singular) fit: see help('isSingular')</span></span>
<span><span class="co">## boundary (singular) fit: see help('isSingular')</span></span>
<span><span class="co">## boundary (singular) fit: see help('isSingular')</span></span></code></pre>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">output</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span>caption <span class="op">=</span> <span class="st">"LMM Outputs"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>LMM Outputs</caption>
<colgroup>
<col width="6%">
<col width="10%">
<col width="18%">
<col width="12%">
<col width="8%">
<col width="7%">
<col width="12%">
<col width="13%">
<col width="10%">
</colgroup>
<thead><tr class="header">
<th align="left">effect</th>
<th align="left">Coefficient</th>
<th align="right">Coefficient Estimate</th>
<th align="right">Standard Error</th>
<th align="right">Statistic</th>
<th align="right">df</th>
<th align="right">Unadj p-value</th>
<th align="left">Taxon</th>
<th align="right">Adj p-value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">fixed</td>
<td align="left">(Intercept)</td>
<td align="right">54874.112</td>
<td align="right">36388.018</td>
<td align="right">1.508027</td>
<td align="right">60.58909</td>
<td align="right">0.1367446</td>
<td align="left">Dolosigranulum</td>
<td align="right">0.2355046</td>
</tr>
<tr class="even">
<td align="left">fixed</td>
<td align="left">(Intercept)</td>
<td align="right">5092.576</td>
<td align="right">4968.070</td>
<td align="right">1.025061</td>
<td align="right">38.93384</td>
<td align="right">0.3116612</td>
<td align="left">Streptococcus</td>
<td align="right">0.4255877</td>
</tr>
<tr class="odd">
<td align="left">fixed</td>
<td align="left">(Intercept)</td>
<td align="right">10777.412</td>
<td align="right">5787.774</td>
<td align="right">1.862100</td>
<td align="right">25.88220</td>
<td align="right">0.0739801</td>
<td align="left">Moraxella</td>
<td align="right">0.1638130</td>
</tr>
<tr class="even">
<td align="left">fixed</td>
<td align="left">(Intercept)</td>
<td align="right">60473.055</td>
<td align="right">16671.751</td>
<td align="right">3.627277</td>
<td align="right">45.00978</td>
<td align="right">0.0007276</td>
<td align="left">Staphylococcus</td>
<td align="right">0.0225553</td>
</tr>
<tr class="odd">
<td align="left">fixed</td>
<td align="left">(Intercept)</td>
<td align="right">45165.789</td>
<td align="right">14562.693</td>
<td align="right">3.101472</td>
<td align="right">62.53418</td>
<td align="right">0.0028879</td>
<td align="left">Corynebacterium</td>
<td align="right">0.0288040</td>
</tr>
<tr class="even">
<td align="left">fixed</td>
<td align="left">(Intercept)</td>
<td align="right">-1417.599</td>
<td align="right">4277.273</td>
<td align="right">-0.331426</td>
<td align="right">34.20857</td>
<td align="right">0.7423446</td>
<td align="left">Haemophilus</td>
<td align="right">0.7935408</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.4.0 (2024-04-24)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: AlmaLinux 8.10 (Cerulean Leopard)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS/LAPACK: FlexiBLAS NETLIB;  LAPACK version 3.11.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: America/New_York</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] emmeans_1.10.3  ggeffects_1.7.0 LegATo_1.1.0   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] splines_4.4.0               tibble_3.2.1               </span></span>
<span><span class="co">##   [3] datawizard_0.12.3           polyclip_1.10-6            </span></span>
<span><span class="co">##   [5] XML_3.99-0.17               rpart_4.1.23               </span></span>
<span><span class="co">##   [7] rex_1.2.1                   lifecycle_1.0.4            </span></span>
<span><span class="co">##   [9] edgeR_4.2.0                 doParallel_1.0.17          </span></span>
<span><span class="co">##  [11] globals_0.16.3              lattice_0.22-6             </span></span>
<span><span class="co">##  [13] MASS_7.3-60.2               insight_0.20.3             </span></span>
<span><span class="co">##  [15] MultiAssayExperiment_1.30.3 backports_1.5.0            </span></span>
<span><span class="co">##  [17] magrittr_2.0.3              limma_3.60.0               </span></span>
<span><span class="co">##  [19] plotly_4.10.4               sass_0.4.9                 </span></span>
<span><span class="co">##  [21] rmarkdown_2.28              jquerylib_0.1.4            </span></span>
<span><span class="co">##  [23] yaml_2.3.10                 askpass_1.2.1              </span></span>
<span><span class="co">##  [25] reticulate_1.38.0           minqa_1.2.8                </span></span>
<span><span class="co">##  [27] RColorBrewer_1.1-3          multcomp_1.4-25            </span></span>
<span><span class="co">##  [29] abind_1.4-8                 zlibbioc_1.50.0            </span></span>
<span><span class="co">##  [31] GenomicRanges_1.56.1        purrr_1.0.2                </span></span>
<span><span class="co">##  [33] BiocGenerics_0.50.0         TH.data_1.1-2              </span></span>
<span><span class="co">##  [35] tweenr_2.0.3                sandwich_3.1-0             </span></span>
<span><span class="co">##  [37] circlize_0.4.16             GenomeInfoDbData_1.2.12    </span></span>
<span><span class="co">##  [39] IRanges_2.38.1              S4Vectors_0.42.1           </span></span>
<span><span class="co">##  [41] ggrepel_0.9.5               rmutil_1.1.10              </span></span>
<span><span class="co">##  [43] inline_0.3.19               listenv_0.9.1              </span></span>
<span><span class="co">##  [45] rentrez_1.2.3               vegan_2.6-6.1              </span></span>
<span><span class="co">##  [47] BiocStyle_2.32.1            umap_0.2.10.0              </span></span>
<span><span class="co">##  [49] RSpectra_0.16-1             spatial_7.3-17             </span></span>
<span><span class="co">##  [51] parallelly_1.38.0           pkgdown_2.0.9              </span></span>
<span><span class="co">##  [53] permute_0.9-7               codetools_0.2-20           </span></span>
<span><span class="co">##  [55] DelayedArray_0.30.1         DT_0.33                    </span></span>
<span><span class="co">##  [57] ggforce_0.4.2               tidyselect_1.2.1           </span></span>
<span><span class="co">##  [59] shape_1.4.6.1               UCSC.utils_1.0.0           </span></span>
<span><span class="co">##  [61] farver_2.1.2                lme4_1.1-35.3              </span></span>
<span><span class="co">##  [63] stable_1.1.6                broom.mixed_0.2.9.5        </span></span>
<span><span class="co">##  [65] matrixStats_1.4.1           stats4_4.4.0               </span></span>
<span><span class="co">##  [67] jsonlite_1.8.9              GetoptLong_1.0.5           </span></span>
<span><span class="co">##  [69] ggalluvial_0.12.5           survival_3.6-4             </span></span>
<span><span class="co">##  [71] iterators_1.0.14            systemfonts_1.0.6          </span></span>
<span><span class="co">##  [73] foreach_1.5.2               tools_4.4.0                </span></span>
<span><span class="co">##  [75] ragg_1.3.0                  Rcpp_1.0.13                </span></span>
<span><span class="co">##  [77] glue_1.8.0                  SparseArray_1.4.8          </span></span>
<span><span class="co">##  [79] BiocBaseUtils_1.6.0         xfun_0.45                  </span></span>
<span><span class="co">##  [81] mgcv_1.9-1                  DESeq2_1.44.0              </span></span>
<span><span class="co">##  [83] MatrixGenerics_1.16.0       GenomeInfoDb_1.40.1        </span></span>
<span><span class="co">##  [85] dplyr_1.1.4                 numDeriv_2016.8-1.1        </span></span>
<span><span class="co">##  [87] withr_3.0.1                 BiocManager_1.30.22        </span></span>
<span><span class="co">##  [89] timeSeries_4032.109         fastmap_1.2.0              </span></span>
<span><span class="co">##  [91] boot_1.3-30                 fansi_1.0.6                </span></span>
<span><span class="co">##  [93] shinyjs_2.1.0               openssl_2.2.1              </span></span>
<span><span class="co">##  [95] animalcules_1.20.0          digest_0.6.37              </span></span>
<span><span class="co">##  [97] R6_2.5.1                    estimability_1.5.1         </span></span>
<span><span class="co">##  [99] textshaping_0.3.7           colorspace_2.1-1           </span></span>
<span><span class="co">## [101] Cairo_1.6-2                 modeest_2.4.0              </span></span>
<span><span class="co">## [103] utf8_1.2.4                  tidyr_1.3.1                </span></span>
<span><span class="co">## [105] generics_0.1.3              data.table_1.16.0          </span></span>
<span><span class="co">## [107] httr_1.4.7                  htmlwidgets_1.6.4          </span></span>
<span><span class="co">## [109] S4Arrays_1.4.1              pkgconfig_2.0.3            </span></span>
<span><span class="co">## [111] gtable_0.3.5                tsne_0.1-3.1               </span></span>
<span><span class="co">## [113] timeDate_4032.109           ComplexHeatmap_2.20.0      </span></span>
<span><span class="co">## [115] covr_3.6.4                  XVector_0.44.0             </span></span>
<span><span class="co">## [117] furrr_0.3.1                 htmltools_0.5.8.1          </span></span>
<span><span class="co">## [119] clue_0.3-65                 scales_1.3.0               </span></span>
<span><span class="co">## [121] Biobase_2.64.0              png_0.1-8                  </span></span>
<span><span class="co">## [123] knitr_1.48                  rstudioapi_0.16.0          </span></span>
<span><span class="co">## [125] reshape2_1.4.4              rjson_0.2.22               </span></span>
<span><span class="co">## [127] nloptr_2.1.1                statip_0.2.3               </span></span>
<span><span class="co">## [129] coda_0.19-4.1               nlme_3.1-164               </span></span>
<span><span class="co">## [131] cachem_1.1.0                zoo_1.8-12                 </span></span>
<span><span class="co">## [133] GlobalOptions_0.1.2         sjlabelled_1.2.0           </span></span>
<span><span class="co">## [135] stringr_1.5.1               parallel_4.4.0             </span></span>
<span><span class="co">## [137] fBasics_4032.96             desc_1.4.3                 </span></span>
<span><span class="co">## [139] pillar_1.9.0                grid_4.4.0                 </span></span>
<span><span class="co">## [141] vctrs_0.6.5                 xtable_1.8-4               </span></span>
<span><span class="co">## [143] cluster_2.1.6               evaluate_1.0.0             </span></span>
<span><span class="co">## [145] magick_2.8.3                mvtnorm_1.2-6              </span></span>
<span><span class="co">## [147] cli_3.6.3                   locfit_1.5-9.10            </span></span>
<span><span class="co">## [149] compiler_4.4.0              rlang_1.1.4                </span></span>
<span><span class="co">## [151] crayon_1.5.3                labeling_0.4.3             </span></span>
<span><span class="co">## [153] plyr_1.8.9                  forcats_1.0.0              </span></span>
<span><span class="co">## [155] fs_1.6.4                    stringi_1.8.4              </span></span>
<span><span class="co">## [157] viridisLite_0.4.2           BiocParallel_1.38.0        </span></span>
<span><span class="co">## [159] TBSignatureProfiler_1.17.0  assertthat_0.2.1           </span></span>
<span><span class="co">## [161] lmerTest_3.1-3              munsell_0.5.1              </span></span>
<span><span class="co">## [163] lazyeval_0.2.2              geepack_1.3.10             </span></span>
<span><span class="co">## [165] Matrix_1.7-0                hms_1.1.3                  </span></span>
<span><span class="co">## [167] stabledist_0.7-1            future_1.34.0              </span></span>
<span><span class="co">## [169] ggplot2_3.5.1               statmod_1.5.0              </span></span>
<span><span class="co">## [171] haven_2.5.4                 SummarizedExperiment_1.34.0</span></span>
<span><span class="co">## [173] highr_0.11                  GUniFrac_1.8               </span></span>
<span><span class="co">## [175] broom_1.0.6                 memoise_2.0.1              </span></span>
<span><span class="co">## [177] bslib_0.8.0                 ape_5.8</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Site created by Aubrey Odom.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
