<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="LegATo">
<title>Introduction to LegATo â€¢ LegATo</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to LegATo">
<meta property="og:description" content="LegATo">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">LegATo</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../index.html">Home</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-tutorials">Tutorials</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-tutorials">
    <h6 class="dropdown-header" data-toc-skip>Command Line Analysis</h6>
    <a class="dropdown-item" href="../articles/LegATo_vignette.html"></a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Function Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Package News</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/aubreyodom/legato">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Introduction to LegATo</h1>
                        <h4 data-toc-skip class="author">Aubrey R.
Odom</h4>
            <address class="author_afil">
      Program in Bioinformatics, Boston University, Boston,
MA<br><a class="author_email" href="mailto:#"></a><a href="mailto:aodom@bu.edu" class="email">aodom@bu.edu</a>
      </address>
                  
            <h4 data-toc-skip class="date">January 25, 2024</h4>
      
      
      <div class="d-none name"><code>LegATo_vignette.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction-to-legato-longitudinal-metagenomic-analysis-toolkit">Introduction to LegATo: Longitudinal mEtaGenomic Analysis
TOolkit<a class="anchor" aria-label="anchor" href="#introduction-to-legato-longitudinal-metagenomic-analysis-toolkit"></a>
</h2>
<div class="section level3">
<h3 id="streamlining-longitudinal-microbiome-profiling-in-bioconductor">Streamlining longitudinal microbiome profiling in Bioconductor<a class="anchor" aria-label="anchor" href="#streamlining-longitudinal-microbiome-profiling-in-bioconductor"></a>
</h3>
<p>LegATo is a suite of open-source software tools for longitudinal
microbiome analysis that is extendable to several different study forms
with optimal ease-of-use for researchers. Microbiome time-series data
presents distinct challenges including complex covariate dependencies
and variety of longitudinal study designs. This toolkit will allow
researchers to determine which microbial taxa are affected over time by
perturbations such as onset of disease or lifestyle choices, and to
predict the effects of these perturbations over time, including changes
in composition or stability of commensal bacteria.</p>
<p>LegATo integrates visualization, modeling and testing procedures. It
will be supplemented by hierarchical clustering tools and multivariate
generalized estimating equations (JGEEs) to adjust for the compositional
nature of microbiome data.</p>
</div>
</div>
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">LegATo</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="data">Data<a class="anchor" aria-label="anchor" href="#data"></a>
</h2>
<p>We will be utilizing nasopharyngeal microbiome data from Zambian
Infants from Odom et al.:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"MAE.RDS"</span>, package <span class="op">=</span> <span class="st">"LegATo"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">dat_0.05</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_animalcules_MAE.html">filter_animalcules_MAE</a></span><span class="op">(</span><span class="va">dat</span>, <span class="fl">0.05</span><span class="op">)</span></span>
<span></span>
<span><span class="va">parsed</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/parse_MAE_SE.html">parse_MAE_SE</a></span><span class="op">(</span><span class="va">dat</span>, which_assay <span class="op">=</span> <span class="st">"MicrobeGenetics"</span>, type <span class="op">=</span> <span class="st">"MAE"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">alluvial_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_long_data.html">get_long_data</a></span><span class="op">(</span><span class="va">dat_0.05</span>, <span class="st">"genus"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Registered S3 method overwritten by 'httr':</span></span>
<span><span class="co">##   method         from  </span></span>
<span><span class="co">##   print.response rmutil</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">group_vars</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HIVStatus"</span>, <span class="st">"MothChild"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/get_summary_table.html">get_summary_table</a></span><span class="op">(</span><span class="va">dat_0.05</span>, <span class="va">group_vars</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   HIVStatus MothChild mean_reads  sd_reads min_reads max_reads num_total</span></span>
<span><span class="co">## 1   Control    Infant  156209.16 408000.52     24176   3016276        68</span></span>
<span><span class="co">## 2   Control    Mother   68648.79  50526.74     14339    168157        19</span></span>
<span><span class="co">## 3       HIV    Infant   90008.20  51395.08     13218    202079        61</span></span>
<span><span class="co">## 4       HIV    Mother   72277.53  48180.28     20118    208029        19</span></span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">best_genus</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_top_taxa.html">get_top_taxa</a></span><span class="op">(</span><span class="va">dat_0.05</span>, <span class="st">"genus"</span><span class="op">)</span></span>
<span><span class="va">best_species</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_top_taxa.html">get_top_taxa</a></span><span class="op">(</span><span class="va">dat_0.05</span>, <span class="st">"species"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">this_palette</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/palettes.html" class="external-link">rainbow</a></span><span class="op">(</span><span class="fl">20</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/plot_stacked_area.html">plot_stacked_area</a></span><span class="op">(</span><span class="va">dat_0.05</span>, <span class="st">"genus"</span>, </span>
<span>                         <span class="st">"HIVStatus"</span>,</span>
<span>                         <span class="st">"timepoint"</span>,</span>
<span>                         palette_input <span class="op">=</span> <span class="va">this_palette</span><span class="op">)</span></span></code></pre></div>
<p><img src="LegATo_vignette_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_stacked_bar.html">plot_stacked_bar</a></span><span class="op">(</span><span class="va">dat_0.05</span>, <span class="st">"genus"</span>, </span>
<span>             <span class="st">"HIVStatus"</span>,</span>
<span>             <span class="st">"timepoint"</span>,</span>
<span>             palette_input <span class="op">=</span> <span class="va">this_palette</span><span class="op">)</span></span></code></pre></div>
<p><img src="LegATo_vignette_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_alluvial.html">plot_alluvial</a></span><span class="op">(</span>dat <span class="op">=</span> <span class="va">dat_0.05</span>, </span>
<span>              taxon_level <span class="op">=</span> <span class="st">"genus"</span>, </span>
<span>              covariate_1 <span class="op">=</span> <span class="st">"HIVStatus"</span>,</span>
<span>              covariate_time <span class="op">=</span> <span class="st">"timepoint"</span>,</span>
<span>              palette_input <span class="op">=</span> <span class="va">this_palette</span>,</span>
<span>              subtitle <span class="op">=</span> <span class="st">"Alluvial plot"</span><span class="op">)</span></span></code></pre></div>
<p><img src="LegATo_vignette_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot_spaghetti.html">plot_spaghetti</a></span><span class="op">(</span>dat <span class="op">=</span> <span class="va">dat_0.05</span>,</span>
<span>               covariate_time <span class="op">=</span> <span class="st">"timepoint"</span>,</span>
<span>               covariate_1 <span class="op">=</span> <span class="st">"HIVStatus"</span>,</span>
<span>               unit_var <span class="op">=</span> <span class="st">"Subject"</span>,</span>
<span>               taxon_level <span class="op">=</span> <span class="st">"genus"</span>,</span>
<span>               which_taxon <span class="op">=</span> <span class="st">"Corynebacterium"</span>,</span>
<span>               palette_input<span class="op">=</span> <span class="va">this_palette</span>,</span>
<span>               title <span class="op">=</span> <span class="st">"Spaghetti Plot"</span>,</span>
<span>               subtitle <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"Infant Age (Days)"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Relative Abundance (log CPM)"</span><span class="op">)</span></span></code></pre></div>
<p><img src="LegATo_vignette_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
<div class="section level3">
<h3 id="gee-modeling">GEE Modeling<a class="anchor" aria-label="anchor" href="#gee-modeling"></a>
</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">output</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/run_gee_model.html">run_gee_model</a></span><span class="op">(</span><span class="va">dat</span>, unit_var <span class="op">=</span> <span class="st">"Subject"</span>,</span>
<span>                        fixed_cov <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HIVStatus"</span>, <span class="st">"MothChild"</span>, <span class="st">"Age"</span><span class="op">)</span>,</span>
<span>                        corstr <span class="op">=</span> <span class="st">"ar1"</span>,</span>
<span>                        plot_out <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                        plotsave_loc <span class="op">=</span> <span class="st">"."</span>,</span>
<span>                        plot_terms <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">output</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   Coefficient Coefficient Estimate Standard Error Statistic Unadj p-value</span></span>
<span><span class="co">## 1 (Intercept)           47024.4796      39794.316 1.3963874    0.23732938</span></span>
<span><span class="co">## 2 (Intercept)            2070.1721       3238.258 0.4086862    0.52263712</span></span>
<span><span class="co">## 3 (Intercept)            9530.8640       5106.186 3.4839455    0.06196683</span></span>
<span><span class="co">## 4 (Intercept)           61209.2478      25927.329 5.5733783    0.01823554</span></span>
<span><span class="co">## 5 (Intercept)           47416.3910      20515.628 5.3417968    0.02082000</span></span>
<span><span class="co">## 6 (Intercept)            -701.1414       1529.727 0.2100794    0.64670520</span></span>
<span><span class="co">##   Lower 95% CI Upper 95% CI           Taxon Adj p-value</span></span>
<span><span class="co">## 1  -30970.9456   125019.905  Dolosigranulum   0.4669326</span></span>
<span><span class="co">## 2   -4276.6975     8417.042   Streptococcus   0.6148113</span></span>
<span><span class="co">## 3    -477.0759    19538.804       Moraxella   0.4669326</span></span>
<span><span class="co">## 4   10392.6166   112025.879  Staphylococcus   0.4669326</span></span>
<span><span class="co">## 5    7206.4993    87626.283 Corynebacterium   0.4669326</span></span>
<span><span class="co">## 6   -3699.3512     2297.068     Haemophilus   0.7164696</span></span></code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Trying out plotting</span></span>
<span><span class="co">#run_gee_model(dat, unit_var = "Subject",</span></span>
<span><span class="co">#              taxon_level = "genus",</span></span>
<span><span class="co">#              fixed_cov = c("MothChild", "HIVStatus", "Age"),</span></span>
<span><span class="co">#              corstr = "ar1",</span></span>
<span><span class="co">#              plot_out = TRUE,</span></span>
<span><span class="co">#              plotsave_loc = #"/restricted/projectnb/infant-microbiome/work/aodom",</span></span>
<span><span class="co">#              plot_terms = "MothChild",</span></span>
<span><span class="co">#              width = 6, height = 4, units = "in", scale = 0.7)</span></span>
<span><span class="co">#</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="add-hotelling">Add hotelling<a class="anchor" aria-label="anchor" href="#add-hotelling"></a>
</h2>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/test_hotelling_t2.html">test_hotelling_t2</a></span><span class="op">(</span>dat <span class="op">=</span> <span class="va">dat_0.05</span>,</span>
<span>                  test_index <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">dat_0.05</span><span class="op">$</span><span class="va">MothChild</span> <span class="op">==</span> <span class="st">"Infant"</span> <span class="op">&amp;</span></span>
<span>                                       <span class="va">dat_0.05</span><span class="op">$</span><span class="va">timepoint</span> <span class="op">==</span> <span class="fl">6</span><span class="op">)</span>,</span>
<span>                  taxon_level <span class="op">=</span> <span class="st">"genus"</span>,</span>
<span>                  <span class="co"># To avoid n &lt; p, use top 5-6 species</span></span>
<span>                  num_taxa <span class="op">=</span> <span class="fl">6</span>,</span>
<span>                  paired <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                  grouping_var <span class="op">=</span> <span class="st">"HIVStatus"</span>,</span>
<span>                  pairing_var <span class="op">=</span> <span class="st">"pairing"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $df1</span></span>
<span><span class="co">## [1] 6</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $df2</span></span>
<span><span class="co">## [1] 2</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $crit_F</span></span>
<span><span class="co">## [1] 19.32953</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $F_stat</span></span>
<span><span class="co">## [1] 2.578132</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $pvalue</span></span>
<span><span class="co">## [1] 0.3056466</span></span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/test_hotelling_t2.html">test_hotelling_t2</a></span><span class="op">(</span>dat <span class="op">=</span> <span class="va">dat_0.05</span>,</span>
<span>                  test_index <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">dat_0.05</span><span class="op">$</span><span class="va">MothChild</span> <span class="op">==</span> <span class="st">"Mother"</span> <span class="op">&amp;</span></span>
<span>                                      <span class="va">dat_0.05</span><span class="op">$</span><span class="va">timepoint</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span>,</span>
<span>                  taxon_level <span class="op">=</span> <span class="st">"genus"</span>,</span>
<span>                  <span class="co"># To avoid n &lt; p, use top 5-6 species</span></span>
<span>                  num_taxa <span class="op">=</span> <span class="fl">12</span>,</span>
<span>                  grouping_var <span class="op">=</span> <span class="st">"HIVStatus"</span>,</span>
<span>                  unit_var <span class="op">=</span> <span class="st">"Subject"</span>,</span>
<span>                  paired <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $df1</span></span>
<span><span class="co">## [1] 12</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $df2</span></span>
<span><span class="co">## [1] 7</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $crit_F</span></span>
<span><span class="co">## [1] 3.574676</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $F_stat</span></span>
<span><span class="co">## [1] 0.4781539</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $pvalue</span></span>
<span><span class="co">## [1] 0.875078</span></span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: AlmaLinux 8.9 (Midnight Oncilla)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /share/pkg.8/r/4.3.1/install/lib64/R/lib/libRblas.so </span></span>
<span><span class="co">## LAPACK: /share/pkg.8/r/4.3.1/install/lib64/R/lib/libRlapack.so;  LAPACK version 3.11.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: America/New_York</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] LegATo_0.99.0    BiocStyle_2.28.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] rstudioapi_0.15.0           jsonlite_1.8.8             </span></span>
<span><span class="co">##   [3] umap_0.2.10.0               MultiAssayExperiment_1.28.0</span></span>
<span><span class="co">##   [5] magrittr_2.0.3              farver_2.1.1               </span></span>
<span><span class="co">##   [7] rmarkdown_2.25              geepack_1.3.9              </span></span>
<span><span class="co">##   [9] fs_1.6.3                    zlibbioc_1.48.0            </span></span>
<span><span class="co">##  [11] ragg_1.2.5                  vctrs_0.6.5                </span></span>
<span><span class="co">##  [13] memoise_2.0.1               RCurl_1.98-1.14            </span></span>
<span><span class="co">##  [15] askpass_1.2.0               BiocBaseUtils_1.4.0        </span></span>
<span><span class="co">##  [17] htmltools_0.5.7             S4Arrays_1.2.0             </span></span>
<span><span class="co">##  [19] broom_1.0.5                 SparseArray_1.2.3          </span></span>
<span><span class="co">##  [21] sass_0.4.8                  TBSignatureProfiler_1.15.0 </span></span>
<span><span class="co">##  [23] bslib_0.6.1                 htmlwidgets_1.6.4          </span></span>
<span><span class="co">##  [25] desc_1.4.3                  plyr_1.8.9                 </span></span>
<span><span class="co">##  [27] covr_3.6.2                  plotly_4.10.2              </span></span>
<span><span class="co">##  [29] cachem_1.0.8                lifecycle_1.0.4            </span></span>
<span><span class="co">##  [31] iterators_1.0.14            pkgconfig_2.0.3            </span></span>
<span><span class="co">##  [33] Matrix_1.6-5                R6_2.5.1                   </span></span>
<span><span class="co">##  [35] fastmap_1.1.1               GenomeInfoDbData_1.2.11    </span></span>
<span><span class="co">##  [37] MatrixGenerics_1.14.0       clue_0.3-65                </span></span>
<span><span class="co">##  [39] digest_0.6.34               colorspace_2.1-0           </span></span>
<span><span class="co">##  [41] spatial_7.3-16              S4Vectors_0.40.2           </span></span>
<span><span class="co">##  [43] DESeq2_1.40.1               RSpectra_0.16-1            </span></span>
<span><span class="co">##  [45] textshaping_0.3.6           GenomicRanges_1.54.1       </span></span>
<span><span class="co">##  [47] vegan_2.6-4                 labeling_0.4.3             </span></span>
<span><span class="co">##  [49] fansi_1.0.6                 httr_1.4.7                 </span></span>
<span><span class="co">##  [51] abind_1.4-5                 mgcv_1.8-42                </span></span>
<span><span class="co">##  [53] compiler_4.3.1              withr_3.0.0                </span></span>
<span><span class="co">##  [55] backports_1.4.1             inline_0.3.19              </span></span>
<span><span class="co">##  [57] BiocParallel_1.36.0         highr_0.10                 </span></span>
<span><span class="co">##  [59] MASS_7.3-60                 tsne_0.1-3.1               </span></span>
<span><span class="co">##  [61] openssl_2.1.1               DelayedArray_0.28.0        </span></span>
<span><span class="co">##  [63] fBasics_4022.94.9000        permute_0.9-7              </span></span>
<span><span class="co">##  [65] tools_4.3.1                 ape_5.7-1                  </span></span>
<span><span class="co">##  [67] rentrez_1.2.3               glue_1.7.0                 </span></span>
<span><span class="co">##  [69] stabledist_0.7-1            nlme_3.1-162               </span></span>
<span><span class="co">##  [71] grid_4.3.1                  cluster_2.1.4              </span></span>
<span><span class="co">##  [73] reshape2_1.4.4              generics_0.1.3             </span></span>
<span><span class="co">##  [75] gtable_0.3.4                tidyr_1.3.0                </span></span>
<span><span class="co">##  [77] data.table_1.14.10          utf8_1.2.4                 </span></span>
<span><span class="co">##  [79] XVector_0.42.0              rmutil_1.1.10              </span></span>
<span><span class="co">##  [81] BiocGenerics_0.48.1         ggrepel_0.9.3              </span></span>
<span><span class="co">##  [83] foreach_1.5.2               pillar_1.9.0               </span></span>
<span><span class="co">##  [85] stringr_1.5.1               limma_3.56.2               </span></span>
<span><span class="co">##  [87] splines_4.3.1               dplyr_1.1.4                </span></span>
<span><span class="co">##  [89] lattice_0.21-8              GUniFrac_1.8               </span></span>
<span><span class="co">##  [91] tidyselect_1.2.0            locfit_1.5-9.8             </span></span>
<span><span class="co">##  [93] knitr_1.45                  bookdown_0.34              </span></span>
<span><span class="co">##  [95] IRanges_2.36.0              edgeR_3.42.4               </span></span>
<span><span class="co">##  [97] SummarizedExperiment_1.32.0 stats4_4.3.1               </span></span>
<span><span class="co">##  [99] xfun_0.41                   Biobase_2.62.0             </span></span>
<span><span class="co">## [101] statmod_1.5.0               timeDate_4032.109          </span></span>
<span><span class="co">## [103] matrixStats_1.2.0           DT_0.28                    </span></span>
<span><span class="co">## [105] rex_1.2.1                   stringi_1.8.3              </span></span>
<span><span class="co">## [107] lazyeval_0.2.2              yaml_2.3.8                 </span></span>
<span><span class="co">## [109] evaluate_0.23               codetools_0.2-19           </span></span>
<span><span class="co">## [111] timeSeries_4030.106.9000    tibble_3.2.1               </span></span>
<span><span class="co">## [113] BiocManager_1.30.22         cli_3.6.2                  </span></span>
<span><span class="co">## [115] rpart_4.1.19                reticulate_1.30            </span></span>
<span><span class="co">## [117] systemfonts_1.0.4           munsell_0.5.0              </span></span>
<span><span class="co">## [119] jquerylib_0.1.4             Rcpp_1.0.12                </span></span>
<span><span class="co">## [121] GenomeInfoDb_1.38.5         stable_1.1.6               </span></span>
<span><span class="co">## [123] png_0.1-8                   XML_3.99-0.16              </span></span>
<span><span class="co">## [125] parallel_4.3.1              modeest_2.4.0              </span></span>
<span><span class="co">## [127] ellipsis_0.3.2              pkgdown_2.0.7              </span></span>
<span><span class="co">## [129] ggplot2_3.4.4               assertthat_0.2.1           </span></span>
<span><span class="co">## [131] animalcules_1.18.2          ggalluvial_0.12.5          </span></span>
<span><span class="co">## [133] bitops_1.0-7                viridisLite_0.4.2          </span></span>
<span><span class="co">## [135] scales_1.3.0                statip_0.2.3               </span></span>
<span><span class="co">## [137] purrr_1.0.2                 crayon_1.5.2               </span></span>
<span><span class="co">## [139] rlang_1.1.3                 shinyjs_2.1.0</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>NULL</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>NULL</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
